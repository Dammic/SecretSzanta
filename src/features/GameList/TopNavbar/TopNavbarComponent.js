import React from 'react'

import { ControlButton } from '../../Shared/Buttons'

import styles from './TopNavbar.css'

export const TopNavbarComponent = ({
    onShowModal,
}) => (
    <div className={styles.topBar}>
        <ControlButton onClick={onShowModal}>New room</ControlButton>
    </div>
)
