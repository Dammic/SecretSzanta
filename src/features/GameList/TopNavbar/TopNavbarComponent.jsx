import React from 'react'

import { Button } from '../../Shared/Buttons'

import styles from './TopNavbar.css'

export const TopNavbarComponent = ({
    onShowModal,
}) => (
    <div className={styles.topBar}>
        <Button onClick={onShowModal}>New room</Button>
    </div>
)
