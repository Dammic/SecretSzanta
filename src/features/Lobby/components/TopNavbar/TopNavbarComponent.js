import React from 'react'
import PropTypes from 'prop-types'
import { Button } from '../../../Shared/Buttons'

import styles from './TopNavbar.css'
import commonStyles from '../../../Shared/CommonStyles/commonStyles.css'

export const TopNavbarComponent = ({ onShowModal, userName }) => (
    <div className={styles.topBar}>
        <span className={commonStyles.ellipsis}>Hello <b>{userName}</b>!</span>
        <div>
            <Button onClick={onShowModal}>New room</Button>
        </div>
    </div>
)


TopNavbarComponent.propTypes = {
    onShowModal: PropTypes.func,
    userName: PropTypes.string,
}
