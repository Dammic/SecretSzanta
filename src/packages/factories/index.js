export { arrayOf } from './arrayOf'
export { aMessage } from './messages'
export { aRoom, aRoomsList } from './lobby'
