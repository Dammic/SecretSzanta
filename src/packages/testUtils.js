import renderer from 'react-test-renderer';
import ShallowRenderer from 'react-test-renderer/shallow';

const shallowRenderer = new ShallowRenderer()

export const expectMatchingSnapshot = (element) => {
    const renderedElement = renderer.create(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}

export const expectShallowMatchingSnapshot = (element) => {
    const renderedElement = shallowRenderer.render(element)
    expect(renderedElement).toMatchSnapshot()
    renderedElement.unmount()
}
