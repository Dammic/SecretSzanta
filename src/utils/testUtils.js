import { mount } from 'enzyme'

export const expectMatchingSnapshot = (element) => {
    const renderedComponent = mount(element)
    expect(renderedComponent).toMatchSnapshot()
}
