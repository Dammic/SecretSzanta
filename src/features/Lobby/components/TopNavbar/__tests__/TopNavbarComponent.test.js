import React from 'react'
import { shallow } from 'enzyme'
import { expectShallowMatchingSnapshot } from 'packages/testUtils'
import { TopNavbarComponent } from '../TopNavbarComponent'

const setupProps = (propsOverrides = {}, renderMethod = shallow) => {
    const props = {
        userName: 'mockUserName',
        onShowModal: jest.fn(),
        ...propsOverrides,
    }
    const component = renderMethod(<TopNavbarComponent {...props} />)
    return { props, component }
}

describe('<TopNavbarComponent />', () => {
    it('matches snapshot', () => {
        const { props } = setupProps()
        expectShallowMatchingSnapshot(<TopNavbarComponent {...props} />)
    })
})
