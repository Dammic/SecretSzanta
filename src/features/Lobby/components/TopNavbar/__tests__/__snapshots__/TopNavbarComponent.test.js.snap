// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopNavbarComponent /> matches snapshot 1`] = `
<div
  className="topBar"
>
  <span
    className="ellipsis"
  >
    Hello 
    <b>
      mockUserName
    </b>
    !
  </span>
  <div>
    <Button
      onClick={[MockFunction]}
    >
      New room
    </Button>
  </div>
</div>
`;
